File: restaurant/state/ready_state.py
from state.dish_state import DishState

class ReadyState(DishState):
    """Trạng thái: Đơn hàng đã sẵn sàng"""
    
    def handle(self, context):
        """Xử lý khi đơn hàng đã được nấu xong"""
        print(f"Đơn hàng {context.order_id} đã sẵn sàng phục vụ.")
        context.state = ReadyState()
    
    def get_status_name(self) -> str:
        return "Ready"

File: backend/src/com/restaurant/pattern/state/ReadyState.java
package com.restaurant.pattern.state;

/**
 * Trạng thái "Sẵn sàng" - Khi món ăn đã sẵn sàng
 */
public class ReadyState implements DishState {
    private OrderItem orderItem;

    public ReadyState(OrderItem orderItem) {
        this.orderItem = orderItem;
    }

    @Override
    public void nextState() {
        orderItem.setState(new ServedState(orderItem));
    }

    @Override
    public String getStatus() {
        return "Sẵn sàng";
    }

    @Override
    public String toString() {
        return getStatus();
    }
}
